modules/nw-about-configuring-master-interface-container.adoc

```diff
@@ -0,0 +1,258 @@
++
+[source,yaml]
+----
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetworkNodePolicy
+metadata:
+ name: sriovnic
+ namespace: openshift-sriov-network-operator
+spec:
+ deviceType: netdevice
+ isRdma: false
+ needVhostNet: true
+ nicSelector:
+   vendor: "15b3" <1>
+   deviceID: "101b" <2>
+   rootDevices: ["00:05.0"]
+ numVfs: 10
+ priority: 99
+ resourceName: sriovnic
+ nodeSelector:
+    feature.node.kubernetes.io/network-sriov.capable: "true"
+----
++
+[NOTE]
+====
+The SR-IOV network node policy configuration example, with the setting `deviceType: netdevice`, is tailored specifically for Mellanox Network Interface Cards (NICs).
+====
++
+<1> The vendor hexadecimal code of the SR-IOV network device. The value `15b3` is associated with a Mellanox NIC.
+<2> The device hexadecimal code of the SR-IOV network device.
+
+.. Apply the YAML by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f sriov-node-network-policy.yaml
+----
++
+[NOTE]
+====
+Applying this might take some time due to the node requiring a reboot.
+====
+
+. Create an SR-IOV network:
+
+.. Create the `SriovNetwork` custom resource (CR) for the additional SR-IOV network attachment as in the following example CR. Save the YAML as the file `sriov-network-attachment.yaml`:
++
+[source,yaml]
+----
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetwork
+metadata:
+ name: sriov-network
+ namespace: openshift-sriov-network-operator
+spec:
+ networkNamespace: test-namespace
+ resourceName: sriovnic
+ spoofChk: "off"
+ trust: "on"
+----
+
+.. Apply the YAML by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f sriov-network-attachment.yaml
+----
+
+. Create a YAML file for the VLAN additional network configuration and then save the YAML in the `vlan100-additional-network-configuration.yaml` file:
++
+[source,yaml]
+----
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: vlan-100
+  namespace: test-namespace
+spec:
+  config: |
+    {
+      "cniVersion": "0.4.0",
+      "name": "vlan-100",
+      "plugins": [
+        {
+          "type": "vlan",
+          "master": "ext0", <1>
+          "mtu": 1500,
+          "vlanId": 100,
+          "linkInContainer": true, <2>
+          "ipam": {"type": "whereabouts", "ipRanges": [{"range": "1.1.1.0/24"}]}
+        }
+      ]
+    }
+----
++
+<1> The VLAN configuration needs to specify the master name. This can be configured in the pod networks annotation.
+<2> The `linkInContainer` parameter must be specified.
+
+. Apply the YAML by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f vlan100-additional-network-configuration.yaml
+----
+
+. Create a pod definition by using the earlier specified networks and then save the YAML in the `pod-a.yaml` file. The pod definition must use the following networks annotation:
++
+[source,yaml]
+----
+apiVersion: v1
```

Comment:
I failed to apply this resource due to:

```
Warning: resource namespaces/test-namespace is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by oc apply. oc apply should only be used on resources created declaratively by either oc create --save-config or oc apply. The missing annotation will be patched automatically.
namespace/test-namespace configured
Error from server (Forbidden): error when creating "./pod-a.yaml": pods "nginx-pod" is forbidden: violates PodSecurity "restricted:v1.24": seccompProfile (pod or container "nginx-container" must set securityContext.seccompProfile.type to "Ru
```ntimeDefault" or "Localhost")

Reply:
I suspect it's because of invalid namespace config. When I change it to:
```
---
apiVersion: v1
kind: Namespace
metadata:
  name: test-namespace
  labels:
    pod-security.kubernetes.io/enforce: privileged
    pod-security.kubernetes.io/audit: privileged
    pod-security.kubernetes.io/warn: privileged
    security.openshift.io/scc.podSecurityLabelSync: "false"
---
```
It works.